### PR TITLE
Preserve filters during navigating back to dashboard

### PR DESCRIPTION
**Pull request by uladzimirdev**
Related to https://github.com/metabase/metabase/issues/31786

### Description

Preserve filters during navigating back to dashboard

### How to verify

There is an e2e test, but manually

1. open dashboard with a question
2. set filter value to any
3. navigate to a question
4. click back button to return back to dashboard
5. filter value should have the previous value

### Demo

https://www.loom.com/share/8b1ebde1f23e428b915e60e9b3c9c59e

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: frontend/src/metabase/hooks/use-web-notification.ts [hunk 1]
```diff
@@ -43,5 +43,5 @@ export function useWebNotification() {
     );
   }, []);
 
-  return [requestPermission, showNotification];
+  return { requestPermission, showNotification };
```
Comment (uladzimirdev) [inline frontend/src/metabase/hooks/use-web-notification.ts:46]: without this change TS couldn't identify type of the function unfortunately

File: frontend/src/metabase/dashboard/components/Dashboard/Dashboard.jsx [hunk 2]
```diff
@@ -186,6 +186,7 @@ class Dashboard extends Component {
     try {
       await fetchDashboard(dashboardId, location.query, {
         clearCache: !isNavigatingBackToDashboard,
+        preserveParameters: isNavigatingBackToDashboard,
```
Comment (uladzimirdev) [inline frontend/src/metabase/dashboard/components/Dashboard/Dashboard.jsx:189]: main change

File: e2e/test/scenarios/dashboard/dashboard-back-navigation.cy.spec.js [hunk 3]
```diff
@@ -256,9 +257,31 @@ describe(
...
+      getDashboardCard().within(() => {
+        cy.findByText("Sleep card").click();
+        cy.wait("@card");
+      });
+
+      filterWidget().findByPlaceholderText("sleep").should("have.value", "5");
+
+      queryBuilderHeader().findByLabelText("Back to Sleep dashboard").click();
+
+      getDashboardCard().within(() => {
```
Comment (ranquild) [inline e2e/test/scenarios/dashboard/dashboard-back-navigation.cy.spec.js:279]: Could you also check that data for the card has been re-requested?

**Review by ranquild — APPROVED** [review]
Locally works correctly

File: e2e/test/scenarios/dashboard/dashboard-back-navigation.cy.spec.js (hunk 3 above)
Comment (uladzimirdev) [reply e2e/test/scenarios/dashboard/dashboard-back-navigation.cy.spec.js:283]: added check on number of requests, didn't find a beautiful way to verify the actual request params

**Pull request merged**